content/strings/KMP.h

```diff
@@ -20,6 +23,15 @@ vi pi(const string& s) {
 	return p;
 }
 
+vi z(const vector<int>& p) {
+	vi Z(sz(p));
+    	rep(i,0,sz(p))
+        	Z[i-p[i]+1] = max(Z[i-p[i]+1], p[i]);
+    	rep(i,0,sz(p)) if (p[i] && Z[p[i] - 1])
+        	Z[i] = max(Z[i], P[Z[P[i] - 1] - 1]);
```

Comment:
is that a capital `P`? (does this compile?)

Reply:
Darnit, I thought my coding-in-the-edit-field would work :(

Reply:
Now add a test for penance!

Reply:
See https://github.com/ludopulles/kactl/commit/e1fea999ec000dd677ec24515361abf844e9ec35#diff-4c06cbaa19a2ea174dc598afa2a9c903 for some  testcode :smile:.